### DISCUSSION TITLE
Does Zed currently support the Local Variables Query feature of tree-sitter?

### DISCUSSION THREAD

**Discussion by ClanEver** [zed-industries/zed#18201] [Feature Requests]
https://tree-sitter.github.io/tree-sitter/syntax-highlighting#local-variables

![image](https://github.com/user-attachments/assets/bd573c14-d2e5-43bc-8252-bd95a4a21b68)

I want to implement this effect (Highlight function parameters separately within the function body) in Zed. Trying to modify the .scm file had no effect.

Does Zed currently support this?

**Comment by notpeter** [comment]
It may not be immediately apparent, but the ruby extension included in the repo will not automatically be used if you build zed from source.  Installing extensions will leverage the public [extension](https://github.com/zed-industries/extension) when installing extensions.  Instead you will need to use "install dev extension" in the extension pane and point it at the `extensions/ruby/` inside the Zed repo.  (You can even do this from a normal release build of Zed).

Once you've gotten your modified version of the extension installed you can open "debug: open syntax tree view from the command palette and see the tree-sitter parse-tree and whether your .scm changes are reflected.

As for whether Zed "supports" this, my guess would be not explicitly (yet). None of the existing grammars use `@local` so there is unlikely to be out of the box theme support to assign a specific colors. Give it a whirl and see if it works. (caveat: I'm no tree-sitter expert)

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
